src/app/psbt.ts

```diff
@@ -375,24 +374,35 @@ const Psbt = function (config) {
             ownedUtxos, destinationBtcAddress, sendFeeRate
         }) => {
             const utxosWithInscription = selectedUtxos.filter(utxo => utxo.inscriptionId);
-            const utxosWithoutInscription = selectedUtxos.filter(utxo => !utxo.inscriptionId);
+            let utxosWithoutInscription = selectedUtxos.filter(utxo => !utxo.inscriptionId);
+
+            // Remove runes if there are inscriptions
+            if (utxosWithInscription.length > 0) {
+                utxosWithoutInscription = await removeRunes(utxosWithoutInscription);
+            }
+
             if (utxosWithInscription.length === 0 && utxosWithoutInscription.length === 0) {
                 throw new Error('At least one ordinal or utxo is required.');
             }
             const provider = SessionStorage.get(SessionsStorageKeys.DOMAIN);
             if (provider !== 'alby') {
                 throw new Error('Signing not supported.');
             }
-
+ 
             // only used if selectedCardinalAmount is not large enough to cover fees
-            const availableCardinalUtxos = ownedUtxos.filter(utxo => !selectedUtxos.some(ownedUtxo => ownedUtxo.txid === utxo.txid))
-                .filter((x) => x.status.confirmed)
-                .filter((x) => x.value > 10000)
-                .filter(utxo => !utxo.inscriptionId)
+            let cardinalUtxos = ownedUtxos
+                .filter(utxo =>
+                    !selectedUtxos.some(ownedUtxo => ownedUtxo.txid === utxo.txid) &&
+                    utxo.status.confirmed &&
+                    utxo.value > 10000 &&
+                    !utxo.inscriptionId
```

Comment:
shouldn't we filter out runes here too?
@topether21

Reply:
oh nevermind i see we do below

Reply:
Just for clarity and readability, I separated the filter.